indico/modules/events/contributions/controllers/management.py

```diff
@@ -459,3 +466,108 @@ def _process(self):
+        custom_field_types = sorted(get_contrib_field_types().values(), key=attrgetter('friendly_name'))
+        return jsonify_template('events/contributions/management/fields_dialog.html', event=self.event_new,
+                                custom_fields=custom_fields, custom_field_types=custom_field_types)
+
+
+class RHSortContributionFields(RHManageContributionsBase):
+    """Sort the custom contribution fields of an event"""
+
+    def _process(self):
+        field_by_id = {field.id: field for field in self.event_new.contribution_fields.all()}
+        field_ids = map(int, request.form.getlist('field_ids'))
+        for index, field_id in enumerate(field_ids, 0):
+            field_by_id[field_id].position = index
+            del field_by_id[field_id]
+        for index, field in enumerate(sorted(field_by_id.values(), key=attrgetter('position')), len(field_ids)):
+            field.position = index
+        db.session.flush()
+        self.event_new.log(EventLogRealm.management, EventLogKind.change, 'Contributions',
+                           'Fields have been reordered', session.user)
+        return jsonify_data(flash=False)
+
+
+class RHCreateContributionField(RHManageContributionsBase):
+    """Dialog to create a new custom field"""
+
+    def _checkParams(self, params):
+        RHManageContributionsBase._checkParams(self, params)
+        field_types = get_contrib_field_types()
+        try:
+            self.field = field_types[request.view_args['field_type']]
+        except KeyError:
+            raise NotFound
+
+    def _process(self):
+        form = self.field.create_config_form()
+        if form.validate_on_submit():
+            contrib_field = ContributionField()
+            field = self.field(contrib_field)
+            print form.data
+            field.update_object(form.data)
+            self.event_new.contribution_fields.append(contrib_field)
+            db.session.flush()
+            self.event_new.log(EventLogRealm.management, EventLogKind.positive, 'Contributions',
+                               'Added field: {}'.format(contrib_field.title), session.user)
+            return jsonify_data(flash=False)
+        return jsonify_template('forms/form_common_fields_first.html', form=form)
+
+
+class RHManageContributionFieldBase(RHManageContributionBase):
+    """Manage a custom contribution field of an event"""
+
+    normalize_url_spec = {
+        'locators': {
+            lambda self: self.contrib_field
+        }
+    }
+
+    def _checkParams(self, params):
+        RHManageContributionsBase._checkParams(self, params)
+        self.contrib_field = ContributionField.get_one(request.view_args['contrib_field_id'])
+
+
+class RHEditContributionField(RHManageContributionFieldBase):
+    """Dialog to edit a custom field"""
+
+    def _process(self):
+        field_class = get_contrib_field_types()[self.contrib_field.field_type]
```

Comment:
`ContributionField` has a property for this. `field_impl` or `field` (check the model, i'm lazy :D)